CONTRIBUTING.md

```diff
@@ -1 +1,101 @@
-# Contributing
+# Contributing 
+
+Thank you for considering contributing to our project! We are grateful for any time and effort you put into making this project better. 
+
+There are several ways you can contribute:
+
+* Submitting bug reports and feature requests
+
+* Submitting pull requests for code changes
+
+* Adding documentation or examples
+
+* Participating in discussions or providing feedback
+
+Before contributing, please read this guide. If you have any questions, feel free to ask in our [Slack](https://join.slack.com/t/gridstatus/shared_invite/zt-1jk6vlzt2-Lzz4pdpjkJYVUJkynOiIvQ) or open a GitHub issue.
+
+## Submitting Bug Reports and Feature Requests
+
+If you find a bug in the code or have an idea for a new feature, please open a new issue and provide as much detail as possible.
+
+## Submitting Pull Requests
+
+We welcome pull requests for code changes! Before starting work on a pull request, please check with a maintainer to ensure that no one else is already working on the same change. Typically, this is done by posting a comment on the appropriate GitHub issue stating that you want to work on it.
+
+When submitting a pull request, please make sure to:
+
+* Write tests to cover any changes you make
+
+* Add any or update docstrings relevant to your change
+
+* Follow the existing code styles in place
+
+* Include a detailed description of the changes you made and why they are necessary
+
+
+
+## Setting up a Development Environment
+
+
+To set up a development environment for this project, you will need to:
+
+* Install Python 3.7 or higher and make sure java is installed on your system
+
+* Clone the repository: `git clone https://github.com/kmax12/gridstatus.git`
```

Comment:
Perhaps something to note later: cloning the repository directly will require forking it later and changing the git remote at the Pull Request step.

Reply:
ahh, good point. will clarify